src/system/SystemLayer.cpp

```diff
@@ -303,6 +303,49 @@ Error Layer::StartTimer(uint32_t aMilliseconds, TimerCompleteFunct aComplete, vo
     return lReturn;
 }
 
+/**
+ * @brief
+ *   This method extends a one-shot timer.
+ *
+ *   @note
+ *       Only a single timer is allowed to be started with the same @a aComplete and @a aAppState
+ *       arguments. If called with @a aComplete and @a aAppState identical to an existing timer,
+ *       the currently-running timer will be extended for a duration of specified period.
+ *
+ *   @param[in]  aMilliseconds Extension time in milliseconds.
+ *   @param[in]  aComplete     A pointer to the function called when timer expires.
+ *   @param[in]  aAppState     A pointer to the application state object used when timer expires.
+ *
+ *   @return CHIP_SYSTEM_NO_ERROR On success.
+ *   @return CHIP_SYSTEM_ERROR_NO_MEMORY If a timer cannot be allocated.
+ *   @return CHIP_SYSTEM_ERROR_UNEXPECTED_STATE If System Layer is not initialized.
+ *   @return Other Value indicating timer failed to start or extend.
+ *
+ */
+Error Layer::ExtendTimer(uint32_t aMilliseconds, Layer::TimerCompleteFunct aOnComplete, void * aAppState)
```

Comment:
Should we also have a version of this that takes a `chip::Callback::Callback`?  Followup, if so.

Reply:
Yes, we can add it later if needed